Source/CarthageKit/BuildSettings.swift

```diff
@@ -292,6 +304,11 @@ public struct BuildSettings {
 		return self["TARGET_BUILD_DIR"]
 	}
 
+	public var shouldBuildForDistribution: Result<Bool, CarthageError> {
```

Comment:
Module stability for Swift binaries:

In order to archive & create xcframework for Swift 5.1+Xcode 11 and above, there are two settings that need to be set:
```
BUILD_LIBRARY_FOR_DISTRIBUTION=NO
```
and
```
SKIP_INSTALL=NO
```

More context for the creation & integration of xcframework (apart from Apple docs) can be found also in here:
https://github.com/bielikb/xcframeworks

Reply:
Im happy to help with this PR, so if there's anything I could help with, please let me know ;)

Reply:
@tmspzz Let me know if you need any help ;)

Reply:
@bielikb Is the direction here clear? First effort is described here 

> Allow packing per platform (i.e iOS + simulator). #2801

and it should be followed by 

> Allow packing all platforms (i.e iOS + TvOS + MacOS ...)

See https://github.com/Carthage/Carthage/issues/2890

Reply:
To support both packings you still need to pass `SKIP_INSTALL=NO` along with `BUILD_LIBRARY_FOR_DISTRIBUTION=NO` when archiving the frameworks. I haven't got across such setup in this PR, but if that's planned/implemented already, feel free to skip this comment ;)

Reply:
`BUILD_LIBRARY_FOR_DISTRIBUTION=YES` should be there.

I don't know about `SKIP_INSTALL=NO`, I don't recall it being mentioned anywhere.

Reply:
Very well then, skip install it is. If you plan to continue please do both "phases" separately.
I.e. first "per platform" then "all pack"